internal/pkg/remote/client_fips_test.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+//go:build requirefips
+
+package remote
+
+import (
+	"context"
+	"fmt"
+	"github.com/elastic/elastic-agent-libs/transport/httpcommon"
+	"github.com/elastic/elastic-agent-libs/transport/tlscommon"
+	"github.com/elastic/elastic-agent/pkg/core/logger/loggertest"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"testing"
+)
+
+func TestClientWithUnsupportedTLSVersions(t *testing.T) {
+	const unsupportedForFIPSErrorMsg = "tls: no supported versions satisfy MinVersion and MaxVersion"
+	const supportedForFIPSErrorMsg = "x509: certificate signed by unknown authority"
```

Comment:
nit: `unsupportedForFIPSErroMsg` vs. `supportedForFIPSErrMsg` took me a bit to understand what the names are supposed to express. What do you think about renaming to be more aligned with the actual errMsg and adding a comment explaining unsupported vs supported versions, and that the unknown authority is the tested error for the supported versions?

Reply:
Please ignore; I was still working on the PR, that's why it was still in draft and not in review 🙂 .